nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/FilterAttribute.java

```diff
@@ -257,17 +215,93 @@ private static String getDelimiter() {
+                context.getProperty(ATTRIBUTE_PATTERN).evaluateAttributeExpressions(flowFile).getValue()
+        );
+    }
+
+    enum FilterMode implements DescribedValue {
+        RETAIN(
+                "Retain",
+                "Retains only the attributes matching the filter, all other attributes are removed."
+        ),
+        REMOVE(
+                "Remove",
+                "Removes the attributes matching the filter, all other attributes are retained."
         );
+
+        private final String value;
+        private final String description;
+
+        FilterMode(final String value, final String description) {
+            this.value = value;
+            this.description = description;
+        }
+
+        @Override
+        public String getValue() {
+            return this.value;
+        }
+
+        @Override
+        public String getDisplayName() {
+            return this.value;
+        }
+
+        @Override
+        public String getDescription() {
+            return this.description;
+        }
+
+        public static FilterMode fromValue(final String value) {
+            return FilterAttribute.enumConstantFromValue(FilterMode.class, value);
+        }
     }
 
-    private enum FilterMode {
-        RETAIN,
-        REMOVE,
+    enum MatchingStrategy implements DescribedValue {
+        ENUMERATION(
+                "Enumerate attributes",
+                "Provides a set of attribute keys to filter for, separated by a comma delimiter ','."
+        ),
+        PATTERN(
+                "Use regular expression",
+                "Provides a regular expression to match keys of attributes to filter for."
+        );
+
+        private final String value;
+        private final String description;
+
+        MatchingStrategy(final String value, final String description) {
+            this.value = value;
+            this.description = description;
+        }
+
+        @Override
+        public String getValue() {
+            return this.value;
+        }
+
+        @Override
+        public String getDisplayName() {
+            return this.value;
+        }
+
+        @Override
+        public String getDescription() {
+            return this.description;
+        }
+
+        public static MatchingStrategy fromValue(final String value) {
+            return FilterAttribute.enumConstantFromValue(MatchingStrategy.class, value);
+        }
     }
 
-    private enum MatchingStrategy {
-        ENUMERATION,
-        REGEX,
+    private static <E extends Enum<E> & DescribedValue> E enumConstantFromValue(final Class<E> enumClass, final String value) {
+        for (E enumConstant : enumClass.getEnumConstants()) {
+            if (enumConstant.getValue().equals(value)) {
+                return enumConstant;
+            }
+        }
+
+        throw new IllegalArgumentException(String.format("Unknown %s value [%s]", enumClass.getSimpleName(), value));
     }
```

Comment:
As mentioned in relation to the property value, switching `getValue()` to return the enum `name()` should obviated the need for this method.

Reply:
I thought about this as well. However this prevents providing more readable names to the users of the processor. 
As far as I know best practice of naming enum entries is using all UPPERCASE. 
Thus, all properties using allowableValues based on an enum would be restricted to be in uppercase. I'm not sure if that's a favourable user experience, additionally it somewhat reveals implementation details. 

In case NIFI-12452 / #8102 gets accepted, this method would be provided "globally", reducing the complexity to implement something like this time and time again. 
Scanning through the code I found multiple places where a method similar to this is used. I was hoping NIFI-12452 could be the basis to simplify and align the implementation at these places.

What are you thoughts on this? Or do you think it's not worth "only" to support more readable names?

In case NIFI-12452 is agreed upon, I would like to backport it into the 1.x line as well and refactor this PR to base on those changes.

Reply:
The changes proposed for NIFI-12452 look useful, so thanks for working on those improvements.

In either case, the question comes down to differentiating between the value itself and the display name. In this scenario, the value could match the enum, such as `RETAIN`, but the display name could be `Retain` for presentation purposes. When you refer to the user experience, are you referencing the UI, or programmatic interaction with the REST API?

Reply:
I was talking about the UI users first and foremost.

I'm somewhat confused though. Due to [this comment](https://github.com/apache/nifi/pull/8064#discussion_r1411195114) I was under the impression one should aim to use the same value for both `value` and `display name` for new implementations, where possible. 
I assumed this would be applicable to `AllowableValue`s as well. 
Or was that comment directed at `PropertyDescriptor`s exclusively?

Reply:
Thanks for the reply @EndzeitBegins, and for referencing the comments in relation to the Property Names.

Although this is focusing on the property value, you are right to reference the same question as it relates to user experience in both cases. That's also a good point about the default value.

On further consideration, maintaining parity between the value and display name of the property value seems the most consistent, so your current approach looks good.

Thanks again for the thoughtful interaction on this question. With this covered, I think the changes should be ready to go, so I will follow up soon.